onnxoptimizer/passes/fuse_cast_into_initializer.h

```diff
@@ -0,0 +1,318 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// ATTENTION: The code in this file is highly EXPERIMENTAL.
+// Adventurous users should note that the APIs will probably change.
+
+#pragma once
+
+// Before:
+//   A is in the initializer list
+//   X = Cast(A)
+//   Z = X + Y
+// After:
+//   B is in the initializer list
+//   Z = B + Y
+//
+
+namespace ONNX_NAMESPACE {
+namespace optimization {
+
+namespace {
+
+enum DataType {
+  FLOAT = 1,
+  UINT8 = 2,
+  INT8 = 3,
+  UINT16 = 4,
+  INT16 = 5,
+  INT32 = 6,
+  INT64 = 7,
+  STRING = 8,       // not supported
+  BOOL = 9,         // not supported
+  FLOAT16 = 10,     // not supported
+  DOUBLE = 11,
+  UINT32 = 12,
+  UINT64 = 13,
+  COMPLEX64 = 14,   // not supported
+  COMPLEX128 = 15,  // not supported
+  BFLOAT16 = 16     // not supported
+};
+
+template <typename T, typename A, typename B>
+void copyData(std::vector<A>& dst, const B* src, const size_t len) {
+    dst.reserve(len);
+    for (size_t i=0; i<len; ++i) {
+        dst.push_back(static_cast<T>(src[i]));
```

Comment:
That's just truncating in the case of ints. Wouldn't normalization be better?

Reply:
Maybe better, but that is not how the ONNX Cast operator is defined.